### PR TITLE
Use Fetcher trait for OCSP fetching

### PR DESCRIPTION
**Pull request by antiphoton**
`sxg_rs` now accepts a dynamic `Fetcher` object, and use it to fetch OCSP from the issuer.

### CODE REVIEW

File: cloudflare_worker/worker/src/index.ts [hunk 1]
```diff
@@ -243,3 +236,20 @@ async function generateSxgResponse(request: Request, payload: Response) {
   );
   return responseFromWasm(sxg);
 }
+
+async function fetcher(request: WasmRequest): Promise<WasmResponse> {
+  const response = await fetch(
+    request.url,
+    {
+      body: new Uint8Array(request.body),
```
Comment (banaag) [inline cloudflare_worker/worker/src/index.ts:244]: Sorry for the naive questions.  Why do you need to do a new Uint8Array here when WasmRequest.body is already a Uint8Array?

File: fastly_compute/src/fetcher.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    Request as FastlyRequest,
+    Response as FastlyResponse,
+};
+use sxg_rs::http::{
+    HttpRequest,
+    HttpResponse,
+    Method,
+};
+
+pub struct FastlyFetcher {
```
Comment (banaag) [inline fastly_compute/src/fetcher.rs:26]: Could you add comments documenting these structs/methods? Thanks.

File: sxg_rs/src/fetcher/js_fetcher.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use async_trait::async_trait;
+use js_sys::Function as JsFunction;
+use wasm_bindgen::JsValue;
+use crate::http::{HttpRequest, HttpResponse};
+use super::Fetcher;
+
+pub struct JsFetcher(JsFunction);
+
```
Comment (banaag) [inline sxg_rs/src/fetcher/js_fetcher.rs:21]: Ditto. Please add comments/docs for structs/methods.

File: cloudflare_worker/worker/src/index.ts (hunk 1 above)
Comment (antiphoton) [reply cloudflare_worker/worker/src/index.ts:244]: Thanks for finding this bug. The `WasmRequest.body` should not be a `Uint8Array`, it is now fixed to `number[]`.

File: sxg_rs/src/fetcher/js_fetcher.rs (hunk 3 above)
Comment (antiphoton) [reply sxg_rs/src/fetcher/js_fetcher.rs:22]: Done

File: fastly_compute/src/fetcher.rs (hunk 2 above)
Comment (antiphoton) [reply fastly_compute/src/fetcher.rs:31]: Done

File: fastly_compute/src/fetcher.rs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+/// A [`Fetcher`] implemented by
+/// [Fastly backend](https://developer.fastly.com/reference/api/services/backend/).
+pub struct FastlyFetcher {
+    backend_name: &'static str,
+}
+
+impl FastlyFetcher {
+    /// Construct a new `FastlyFetcher` from the backend name.
```
Comment (banaag) [inline fastly_compute/src/fetcher.rs:36]: Construct -> Constructs (https://doc.rust-lang.org/1.0.0/style/style/comments.html)

File: sxg_rs/src/fetcher/js_fetcher.rs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+use js_sys::Function as JsFunction;
+use wasm_bindgen::JsValue;
+use crate::http::{HttpRequest, HttpResponse};
+use super::Fetcher;
+
+/// A [`Fetcher`] implemented by JavaScript.
+pub struct JsFetcher(JsFunction);
+
+impl JsFetcher {
+    /// Construct a new `JsFetcher` with a given JavaScript function,
```
Comment (banaag) [inline sxg_rs/src/fetcher/js_fetcher.rs:24]: Construct -> Constructs

**Review by banaag — APPROVED** [review]

File: sxg_rs/src/fetcher/js_fetcher.rs (hunk 5 above)
Comment (antiphoton) [reply sxg_rs/src/fetcher/js_fetcher.rs]: Done

File: fastly_compute/src/fetcher.rs (hunk 4 above)
Comment (antiphoton) [reply fastly_compute/src/fetcher.rs]: Done

**Pull request merged**